source/code/source/playfab/PlayFabEventBuffer.cpp

```diff
@@ -133,7 +133,7 @@ namespace PlayFab
         return EventConsumingResult::Success;
     }
 
-    PlayFabEventPacket* PlayFabEventBuffer::CreateEventPacket(uint8_t *location, const uint64_t index, std::shared_ptr<const IPlayFabEmitEventRequest> request)
+    PlayFabEventPacket* PlayFabEventBuffer::CreateEventPacket(uint8_t *location, const uint64_t index, const std::shared_ptr<const IPlayFabEmitEventRequest>& request)
```

Comment:
We should never std::move a shared ptr. We should only move uniques.
This one should probably be a unique, and thus, not const&

Reply:
This gets a bit hairy when we swap to a unique ptr (this change trickles up to the PlayFabEventsApi::EmitEvent public function which will take me a bit to get a valid compile). The arguments passed to this function throughout this file is a std::move'd shared_ptr, which is a public function signature for TryPut (with that const &). SINCE Put is supposed to be taking ownership of the object, it makes sense TO make it something like a unique_ptr all the way through (but I would like to make sure it doesn't require the refactor we want to avoid).